screens/ChatScreen.js

```diff
@@ -42,126 +44,249 @@ class ChatScreen extends Component {
+    };
 
-    console.log("Done mapping state"); */
+    this.updateLocalMessageState(messages);
   }
 
-  async onSend(messages = []) {
-    const promises = messages.map(msg => new Promise((resolve) => {
-      this.channel.sendUserMessage(msg.text, resolve);
-    }));
-    await Promise.all(promises);
+  async updateLocalMessageState(messages) {
+    const totalMessages = messages.items.length;
+    const localMessages = [];
+
+    for (let i = 0; i < totalMessages; i += 1) {
+      const message = messages.items[i];
+      const currMessage = {
+        _id: `${message.index}`,
+        text: `${message.body}`,
+        createdAt: `${message.timestamp}`,
+        user: {
+          _id: `${message.author}`,
+        },
+      };
+
+      localMessages.push(currMessage);
+    }
+
     this.setState(previousState => ({
-      messages: previousState.messages.concat(messages),
+      messages: previousState.messages.concat(localMessages),
     }));
   }
 
-  async onReceive(messages) {
-    console.log('onReceive');
-    this.setState(previousState => ({
-      messages: previousState.messages.concat(messages),
-    }));
-    console.log('onReceive done');
+  async createChannelWithUser(chatClient) {
+    console.log('Creating channel...');
+
+    const user = this.props.navigation.state.params.mentee._id;
+    // Sort the IDs so that we have a deterministic order
+    const channelName = this.getChannelName();
+    return chatClient.createChannel({
+      uniqueName: channelName,
+      friendlyName: channelName,
+    }).then((channel) => {
+      console.log('Created channel');
+      channel.join().catch((err) => {
+        console.error(
+          `Couldn't join channel ${channel.friendlyName} because ${err}`
+        );
+      });
+
+      // Invite other user to your channel
+      channel.invite(user).then(() => {
+        console.log('Your friend has been invited!');
+      }).catch((error) => {
+        console.log(`Couldn't invite user: ${error}`);
+      });
+    }).catch((error) => {
+      console.log(`Error in creating channel: ${error}`);
+    });
   }
 
-  getTwilioToken = async (localToken, userId) => {
-    console.log('Getting Twilio data');
-    console.log(localToken);
-    console.log(userId);
-    const API_URL = 'http://ppeheymentor-env.qhsppj9piv.us-east-2.elasticbeanstalk.com';
+  async initChatClient(token) {
+    console.log('initChatClient');
+
+    TwilioChatClient.create(token, { logLevel: 'info' }).then((chatClient) => {
+      this.client = chatClient;
+
+      this.client.on('tokenAboutToExpire', () => {
+        this.getTwilioToken()
+          .then(newData => this.client.updateToken(newData))
+          .catch((err) => {
+            console.log(`Error getting token on refresh: ${err}`);
+          });
+      });
+
+      // Listen for new invitations to your Client
+      this.client.on('channelInvited', (channel) => {
+        console.log(`Invited to channel ${channel.friendlyName}`);
```

Comment:
A lot of this just seems like logging to the console, which the user can't see anyway. Should we move this to some sort of notification framework, or remove these logs?

Reply:
Yeah this was just for me testing. Removed.